ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py

```diff
@@ -125,19 +124,13 @@ def ethernet_interfaces(self) -> list | None:
 
             # Remove None values
             ethernet_interface = {key: value for key, value in ethernet_interface.items() if value is not None}
-
-            if (found_eth_interface := get_item(ethernet_interfaces, "name", ethernet_interface["name"])) is None:
-                ethernet_interfaces.append(ethernet_interface)
-            else:
-                if found_eth_interface == ethernet_interface:
-                    # Same ethernet_interface information twice in the input data. So not duplicate interface name.
-                    continue
-
-                raise AristaAvdError(
-                    f"Duplicate interface name {ethernet_interface['name']} found while generating ethernet_interfaces for underlay peer:"
-                    f" {ethernet_interface['peer']}, peer_interface: {ethernet_interface['peer_interface']}. Description on duplicate interface name:"
-                    f" {found_eth_interface['description']}"
-                )
+            append_if_not_duplicate(
+                list_of_dicts=ethernet_interfaces,
+                primary_key="name",
+                new_dict=ethernet_interface,
+                context="Ethernet Interfaces defined under underlay",
```

Comment:
There is not really a concept of "under underlay" for a user of AVD so we need to reword this

Reply:
fixed